aptos-move/block-executor/src/errors.rs

```diff
@@ -17,14 +17,39 @@ pub enum IntentionalFallbackToSequential {
     /// We defensively check resource group serialization error in the commit phase.
     /// TODO: should trigger invariant violation in the transaction itself.
     ResourceGroupSerializationError(String),
+    /// Execution of a thread yields a non-recoverable error from the VM. It it happens
+    /// during parallel execution, an error will be logged and sequential fallback taken.
+    FatalVMError(String),
+}
+
+pub(crate) fn log_panic_or(error: &PanicOr<IntentionalFallbackToSequential>, context: &str) {
+    match error {
+        PanicOr::Or(e) => match e {
+            IntentionalFallbackToSequential::ModulePathReadWrite => {
+                debug!("[Execution]: Module read & written {}", context);
+            },
+            IntentionalFallbackToSequential::ResourceGroupSerializationError(msg) => {
+                error!(
+                    "[Execution]: ResourceGroupSerializationError {}, {}",
+                    msg, context
+                );
+            },
+            IntentionalFallbackToSequential::FatalVMError(msg) => {
+                error!("[Execution]: FatalVMError {:?}, {}", msg, context);
+            },
+        },
+        PanicOr::CodeInvariantError(msg) => {
+            error!("[Execution]: CodeInvariantError({:?}), {}", msg, context);
+        },
+    }
 }
 
 #[derive(Clone, Debug, PartialEq, Eq)]
 pub enum BlockExecutionError<E> {
```

Comment:
we can leave for separate PR, but should we just have a different Error being returned by parallel and sequential execution, and removing this error type completely?

that way parallel execution cannot accidentally return BlockExecutionError::FatalVMError, which it never should

Reply:
good point, I added unreachables, but that's much cleaner. Let's do in a separate PR though, might add too much noise here.

Reply:
I can try to make the change and see

Reply:
made some steps in that direction.

Reply:
This is now super easy to check because it literally creates an error at return.